sunpy/net/jsoc/jsoc.py

```diff
@@ -335,8 +336,8 @@ def request_data(self, jsoc_response, method='url', **kwargs):
 
         requests = []
         self.query_args = jsoc_response.query_args
-        supported_protocols = {'fits', 'as-is'}
-        supported_methods = {'url-tar', 'url', 'url-quick'}
+        supported_protocols = sorted({'fits', 'as-is'})
+        supported_methods = sorted({'url-tar', 'url', 'url-quick'})
```

Comment:
If we care about these being sorted, why not just use a list? Taking a set and then converting it to a generator seems insane.

Reply:
Fixed.